useragent/useragent.go

```diff
@@ -43,3 +43,18 @@ func UserAgent(binaryNameCapitalized string, version, commit, buildTime string,
 	builder.WriteByte(')')
 	return builder.String()
 }
+
+func UserAgentWithBeatTelemetry(binaryNameCapitalized string, version string, mode string, unprivileged string) string {
+	var builder strings.Builder
+	builder.WriteString("Elastic-" + binaryNameCapitalized + "/" + version + " ")
+	uaValues := []string{
+		runtime.GOOS,
+		runtime.GOARCH,
+		mode,
+		unprivileged,
+	}
+	builder.WriteByte('(')
+	builder.WriteString(strings.Join(uaValues, "; "))
+	builder.WriteByte(')')
+	return builder.String()
```

Comment:
Since the input is still quite flexible, what I'd suggest is to have this function return an error when the user agent is over 100 bytes. The test doesn't catch all instances of the user agent being too long because that depends on the input.

Then, when you call it you can look at the error and log it but always set the user agent anyway, and the test can just check the function doesn't return an error.